metricq_source_http/source/main.py

```diff
@@ -102,76 +101,83 @@ async def query_data(metric_name, conf):
     Returns:
         tuple -- (metric_name, timestamp of query, queried value(no value == NaN))
     """
-    url = urljoin(conf['host_infos']['host_url'], conf['path'])
+    url = urljoin(conf["host_infos"]["host_url"], conf["path"])
     value = NaN
     ts = metricq.Timestamp.now()
-    if not conf['host_infos']['login_data']['authorized']:
+    if not conf["host_infos"]["login_data"]["authorized"]:
         return metric_name, ts, value
     data = None
-    if ['use_cache'] and (url in cache):
-       data = cache[url]
-    else:
-        try:
-            response = await conf['host_infos']['session'].get(url)
-            if response.status >= 400:
+
+    # I thought about not using a context manager here, so when the cache
+    # isn't used, we don't lock. But decided against it, so I can be sure
+    # there will never be a dangling lock caused by some random exception.
+    # Maybe some day, a python wizard comes by and brings a magic solution
+    # which uses context managers and is conditional. I don't posses such
+    # magic. Maybe vtti the grey will.
```

Comment:
https://docs.python.org/3.7/library/contextlib.html#contextlib.nullcontext

You're welcome